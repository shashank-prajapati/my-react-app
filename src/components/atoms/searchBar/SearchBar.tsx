import React from "react";
import TextField from "@mui/material/TextField";
import SearchIcon from "@mui/icons-material/Search";
import InputAdornment from "@mui/material/InputAdornment";
import { theme } from "../../../themes/themes";


const SearchBar = () => {
  return (
    <div>
      <TextField
        placeholder="Search by title or author"
        variant="standard"
        sx={{
          ".MuiInput-root": {
            fontSize: `${theme.typography.h3.fontSize}`,
          },
          ".MuiInput-root:before,.MuiInput-root:after,.MuiInput-root:hover,.MuiInput-root:hover:not(.Mui-disabled):before":
            {
              borderColor: "extraTextColor.main",
              borderWidth: 2,
            },
          width: "63%",
        }}
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <SearchIcon sx={{ color: "extraTextColor.main" }} />
            </InputAdornment>
          ),
        }}
      />
    </div>
  );
};

export default SearchBar;
