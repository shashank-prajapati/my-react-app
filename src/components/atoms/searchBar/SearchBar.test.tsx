import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SearchBar from './SearchBar';


describe('Testing Search Bar', ()=>{
    it('should match the default snapshot',()=>{
        const {container} = render(<SearchBar/>,);
        expect(container).toMatchSnapshot();
    })
})

