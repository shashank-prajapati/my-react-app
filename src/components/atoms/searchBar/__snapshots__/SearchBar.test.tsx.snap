// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search Bar should match the default snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1e7icfb-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1f67d23-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <input
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-1"
          placeholder="Search by title or author"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
